### 소득 없이 매출액 총합 회귀분석

In [30]:
#상권(nanX,매출액총합,소득O):sang_income_nan.csv
s_df = pd.read_csv("sang_income_nan.csv", encoding='euc-kr')
s_df=s_df[s_df.columns[:-2]]
s_df.columns= ['당월_매출_금액', '기준_분기_코드', '상권_코드', '서비스_업종_코드', '총_직장_인구_수',
       'employee_rate', '집객시설_수', '총_유동인구_수', 'floating_pop_rate',
       '운영_영업_개월_평균', '폐업_영업_개월_평균', '총_상주인구_수', '총_가구_수', '점포_수',
       '유사_업종_점포_수']

In [31]:
s_df.columns

Index(['당월_매출_금액', '기준_분기_코드', '상권_코드', '서비스_업종_코드', '총_직장_인구_수',
       'employee_rate', '집객시설_수', '총_유동인구_수', 'floating_pop_rate',
       '운영_영업_개월_평균', '폐업_영업_개월_평균', '총_상주인구_수', '총_가구_수', '점포_수',
       '유사_업종_점포_수'],
      dtype='object')

In [32]:
for i in s_df[s_df.columns[4:]].columns:
    print(i, end="+")

총_직장_인구_수+employee_rate+집객시설_수+총_유동인구_수+floating_pop_rate+운영_영업_개월_평균+폐업_영업_개월_평균+총_상주인구_수+총_가구_수+점포_수+유사_업종_점포_수+

In [40]:
#robust scaling하기
from sklearn.preprocessing import RobustScaler

dfX=s_df[s_df.columns[4:]]
rb = RobustScaler()
rb.fit(dfX)
X_robust_scaled = rb.transform(dfX)
dfX2=pd.DataFrame(X_robust_scaled, columns= s_df.columns[4:])
s_df_scaled = pd.concat([s_df[s_df.columns[[0,3]]],dfX2], axis=1)

#회귀분석
model = sm.OLS.from_formula("당월_매출_금액 ~ 총_직장_인구_수+employee_rate+집객시설_수+총_유동인구_수+\
운영_영업_개월_평균+폐업_영업_개월_평균+총_상주인구_수+총_가구_수+점포_수+유사_업종_점포_수+C(서비스_업종_코드)+0",data= s_df_scaled)

result = model.fit()
print(result.summary())

                            OLS Regression Results                            
Dep. Variable:               당월_매출_금액   R-squared:                       0.439
Model:                            OLS   Adj. R-squared:                  0.439
Method:                 Least Squares   F-statistic:                     1181.
Date:                Mon, 02 Dec 2019   Prob (F-statistic):               0.00
Time:                        19:59:30   Log-Likelihood:            -1.7290e+06
No. Observations:               81553   AIC:                         3.458e+06
Df Residuals:                   81498   BIC:                         3.459e+06
Df Model:                          54                                         
Covariance Type:            nonrobust                                         
                             coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------------------
C(서비스_업종_코드)[CS100001]  2.34

### 결론
- 다중공선성 문제 발생
- 서비스 업종 코드가 너무 많아서 발생한 것으로 보임
- 상권코드와 마찬가지로 변수를 줄일 필요가 있음 
- 다만 상권코드 R-square 값(0.388)보다 높은 (0.439)가 나왔다는 점이 눈에 띈다